### PR TITLE
chore: Rename Contains to IsSupersetOf

### PR DESCRIPTION
**Pull request by varungandhi-src**
The name Contains is also commonly used for checking if a
set contains an element in other languages, so it leads to
confusing auto-complete. Additionally, IsSupersetOf sounds
more expensive, which may be the case for this operation.

## Test plan

Covered by existing tests

## Changelog

### CODE REVIEW

File: internal/collections/set_test.go [hunk 1]
```diff
@@ -65,16 +65,16 @@ func TestSet(t *testing.T) {
...
-	t.Run("Contains returns true if set contains the other set", func(t *testing.T) {
-		require.True(t, a.Contains(NewSet(1, 2)))
-		require.True(t, a.Contains(NewSet(1, 2, 3)))
-		require.False(t, a.Contains(b))
+	t.Run("IsSupersetOf returns true if set contains the other set", func(t *testing.T) {
+		require.True(t, a.IsSupersetOf(NewSet(1, 2)))
+		require.True(t, a.IsSupersetOf(NewSet(1, 2, 3)))
+		require.False(t, a.IsSupersetOf(b))
 
 		// set always contains self
```
Comment (burmudar) [inline internal/collections/set_test.go:73]:
```suggestion
		// a set is a superset of itself
```

File: internal/collections/set_test.go [hunk 2]
```diff
@@ -65,16 +65,16 @@ func TestSet(t *testing.T) {
...
-		require.True(t, a.Contains(a))
+		require.True(t, a.IsSupersetOf(a))
 
 		// empty set is always contained
```
Comment (burmudar) [inline internal/collections/set_test.go:76]:
```suggestion
		// a set is always the superset of an empty set
```

**Review by burmudar — APPROVED** [review]

**Comment by unknwon** [comment]
not a native speaker... "A contains B" to me means `A >= B`, "A is superset of B" means `A > B`, which changes the semantic meaning to me. Or does "superset" also implies `>=` relation?

**Comment by varungandhi-src** [comment]
In usual math terminology, "superset" includes the equals case, yes.

Most math textbooks use ⊇ for superset and ⊃ for strict superset (at least one element is different). Sometimes ⊃ is used for superset and ⊋ is used for strict superset. 

[Relevant section on Wikipedia](https://en.wikipedia.org/wiki/Subset#%E2%8A%82_and_%E2%8A%83_symbols)

**Comment by unknwon** [comment]
> In usual math terminology, "superset" includes the equals case, yes.
> 
> Most math textbooks use ⊇ for superset and ⊃ for strict superset (at least one element is different). Sometimes ⊃ is used for superset and ⊋ is used for strict superset.
> 
> [Relevant section on Wikipedia](https://en.wikipedia.org/wiki/Subset#%E2%8A%82_and_%E2%8A%83_symbols)

TIL, thank you!

**Pull request merged**